### DISCUSSION TITLE
Teslamate on synology

### DISCUSSION THREAD

**Discussion by fjgaspar** [teslamate-org/teslamate#2968] [Q&A]
Hi,

I have installed teslamate in a synology DS920+ via docker, and using docker composer. This is the yml I have used:

version: "3"

services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: unless-stopped
    environment:
      - ENCRYPTION_KEY=somepass
      - DATABASE_USER=teslamate
      - DATABASE_PASS=somepass
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
      - MQTT_HOST=mosquitto
    ports:
      - 4000:4000
    volumes:
      - /volume1/docker/teslamate/import:/opt/app/import
    cap_drop:
      - all
   
  database:
    image: postgres:14
    restart: always
    environment:
      - POSTGRES_USER=teslamate
      - POSTGRES_PASSWORD=somepass
      - POSTGRES_DB=teslamate
    volumes:
      - teslamate-db:/var/lib/postgresql/data

  grafana:
    image: teslamate/grafana:latest
    restart: always
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=somepass
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
    ports:
      - 3000:3000
    volumes:
      - teslamate-grafana-data:/var/lib/grafana

  mosquitto:
    image: eclipse-mosquitto:2
    restart: always
    command: mosquitto -c /mosquitto-no-auth.conf
    # ports:
    #   - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

volumes:
  teslamate-db:
  teslamate-grafana-data:
  mosquitto-conf:
  mosquitto-data:

It was working correctly and one day, I think it was after a reboot of synology, suddenly I had lost all data. 

I am not an expert in docker, but I have read something about that the processes that run in docker occurs inside the container, so all data generated runs only inside the container. If container stops and starts again, it runs a new instance of the image, and data is lost as it is a different instance (container). To avoid that, it is needed to put the data outside the container, in a shared folder of the Synology nas, i.e what it does the lines

  volumes:
      - /volume1/docker/teslamate/import:/opt/app/import

which redirects to a NAS folder called teslamate/import what it is inside the container folder /opt/app/import.

That is easy. But when it comes to the directories:
volumes:
      - teslamate-db:/var/lib/postgresql/data
volumes:
      - teslamate-grafana-data:/var/lib/grafana
 volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

I cant figure out how to do it. As much as I can understand, a volume called teslamate-db (the database used for store historical data) points to the /var/lib/postgresql/data folder inside the container posrgres, and I don't know how to map it to a folder like /volume1/docker/teslamate/teslamatedb for example, so that if the container reboots, data is on the NAS, not in the container.

I have tried to stop the container, edit it, go to the volumes tab and add and map there, but I get the message: this operation could not be done probably because the network is unstable or system is occupied, try it again later.

Any help would be appreciated.

**Comment by baylanger** [comment]
```
# docker volume list|grep tesla
local     teslamate_mosquitto-conf
local     teslamate_mosquitto-data
local     teslamate_teslamate-db
local     teslamate_teslamate-grafana-data

# docker volume inspect teslamate_teslamate-db
[
    {
        "CreatedAt": "2023-01-06T17:56:21-05:00",
        "Driver": "local",
        "Labels": {
            "com.docker.compose.project": "teslamate",
            "com.docker.compose.version": "1.28.5",
            "com.docker.compose.volume": "teslamate-db"
        },
        "Mountpoint": "/volume1/@docker/volumes/teslamate_teslamate-db/_data",
        "Name": "teslamate_teslamate-db",
        "Options": null,
        "Scope": "local"
    }
```

This gives you the location. In the case above, the volume (data) is not destroy if you delete a container.

You could stop your current containers, create a new one using a different name but with the same config. If that works, then delete this one and recreate the original one.
